conans/test/functional/corrupted_packages_test.py

```diff
@@ -0,0 +1,160 @@
+import os
+import textwrap
+import unittest
+
+from parameterized import parameterized
+
+from conans.model.ref import ConanFileReference, PackageReference
+from conans.test.utils.tools import TestClient, TestServer, NO_SETTINGS_PACKAGE_ID
+
+
+class CorruptedPackagesTest(unittest.TestCase):
+    """
+    Simulate a connection failure or file corruption in the server with missing files for a
+    package and make sure the search, install are possible. Check re-upload is always possible
+    even if the package in the server is not accessible
+    """
+
+    def _setUp(self, revisions_enabled):
+        self.server = TestServer([("*/*@*/*", "*")], [("*/*@*/*", "*")])
+        self.client = TestClient(servers={"default": self.server},
+                                 revisions_enabled=revisions_enabled)
+        conanfile = textwrap.dedent("""
+        from conans import ConanFile
+
+        class Pkg(ConanFile):
+            pass
+        """)
+        self.client.save({"conanfile.py": conanfile})
+        self.client.run("create . Pkg/0.1@user/testing")
+        self.client.run("upload * --all --confirm -r default")
+        # Check files are uploded in this order: conan_package.tgz, conaninfo.txt, conanmanifest.txt
+        order1 = str(self.client.out).find("Uploading conan_package.tgz")
+        order2 = str(self.client.out).find("Uploading conaninfo.txt", order1)
+        order3 = str(self.client.out).find("Uploading conanmanifest.txt", order2)
+        self.assertTrue(order1 < order2 < order3)
+        rrev = "210a4a16419aae28fea1f268a8e4f3d4" if revisions_enabled else "0"
+        pref_str = "Pkg/0.1@user/testing#%s" % rrev
+        prev = "11de80325e8db78617b05384825c6409" if revisions_enabled else "0"
+        self.pref = pref = PackageReference(ConanFileReference.loads(pref_str),
+                                NO_SETTINGS_PACKAGE_ID, prev)
+        self.manifest_path = self.server.server_store.get_package_file_path(pref,
+                                                                            "conanmanifest.txt")
+        self.info_path = self.server.server_store.get_package_file_path(pref, "conaninfo.txt")
+        self.tgz_path = self.server.server_store.get_package_file_path(pref, "conan_package.tgz")
+
+    def _assert_all_package_files_in_server(self):
+        self.assertTrue(os.path.exists(self.manifest_path))
+        self.assertTrue(os.path.exists(self.info_path))
+        self.assertTrue(os.path.exists(self.tgz_path))
+
+    @parameterized.expand([(True,), (False,)])
```

Comment:
I think you don't need the "revisions"/"no revisions" parametrization in this test. It will be run with and without revisions in the complete build, automatically.